backend/tests/integration/accounts/test_views.py

```diff
@@ -19,3 +25,84 @@ def test_me_returns_user_data_if_logged_in(api_client, create_user):
     expected = {"data": {"email": user.email, "roles": ["user"]}}
     response = api_client.get("/accounts/me", content_type="application/json")
     assert response.json == expected
+
+
+def test_login_success_returns_the_correct_template(api_client, create_user):
+    user = create_user(role="user")
+    login_user_via_session(api_client, email=user.email)
+    payload = {"data": {"email": user.email, "roles": ["user"]}}
+
+    expected = render_template("accounts/postmessage.html", payload=payload)
+    response = api_client.get("/accounts/login_success")
+    assert expected == response.get_data(as_text=True)
+
+
+def test_sign_up_required_returns_the_correct_template(api_client, create_user):
+    user = create_user(role="user")
+    login_user_via_session(api_client, email=user.email)
+    payload = {"user_needs_sign_up": True}
+
+    expected = render_template("accounts/postmessage.html", payload=payload)
+    response = api_client.get("/accounts/signup")
+    assert expected == response.get_data(as_text=True)
+
+
+@mock.patch("flask_login.utils._get_user")
+def test_sign_up_user_success(mock_current_user, api_client, create_user):
+    """It's mocking current user because invenio handlers need a lot of things to
+    setup in order to make it properly work and we don't want to test this functinality."""
+
+    def return_true():
+        return True
+
+    current_oauthclient.signup_handlers["orcid"] = {"view": return_true}
```

Comment:
maybe it's better to also "teardown" this in order not to affect other test cases, I see that right now the other tests are overriding this, but it might be confusing when we start adding new tests.